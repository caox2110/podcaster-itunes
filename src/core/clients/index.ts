export { default as HttpClientFactory } from './HttpClient.factory';
export * from './enums';
export * from './types';
