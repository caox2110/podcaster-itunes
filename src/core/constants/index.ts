import { appConstant } from './app.constant';
import { themeConstant } from './theme.constant';
import { urlConstant } from './url.constant';
import { queryClientConstant } from './queryClient.constant';

/**
 * Constants for the app
 *
 * @type {{ app: any; theme: any; url: any; }}
 */
const constants = {
  app: appConstant,
  theme: themeConstant,
  url: urlConstant,
  queryClient: queryClientConstant,
};

if (appConstant.IS_DEV) {
  console.warn('Constants are', constants);
  console.warn(`ENV is ${appConstant.MODE}`);
}

export { constants };
