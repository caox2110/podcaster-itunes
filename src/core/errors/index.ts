export { default as ExceptionHandler } from './Exceptions.handler';
