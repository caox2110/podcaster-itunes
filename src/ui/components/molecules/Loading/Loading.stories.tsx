import type { ComponentStory, ComponentMeta } from '@storybook/react';

import { withVerticalCentered } from '../../../../../.storybook/preview';

import Loading, { LoadingTypeEnum } from '.';

const Template: ComponentStory<typeof Loading> = (properties) => <Loading {...properties} />;

const LoadingLinearProgress = Template.bind({});
LoadingLinearProgress.args = {};

const LoadingCircularProgress = Template.bind({});
LoadingLinearProgress.args = { type: LoadingTypeEnum.circular };

export { LoadingLinearProgress, LoadingCircularProgress };
export default {
  title: 'UI/Molecules/Loading',
  component: Loading,
  decorators: [withVerticalCentered],
  argTypes: {
    type: {
      control: false,
    },
  },
} as ComponentMeta<typeof Loading>;
