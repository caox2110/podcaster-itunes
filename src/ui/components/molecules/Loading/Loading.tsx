import { MuiContainer, MuiLinearProgress, MuiCircularProgress } from '../..';

/**
 * Loading Types Options
 *
 * @enum {number}
 */
enum LoadingTypeEnum {
  circular = 'circular',
  linear = 'linear',
}

type LoadingProperties = { type?: LoadingTypeEnum };

/**
 * Loading Component definition.
 *
 * @param {{ type?: LoadingTypeEnum }} { type, ...rest }
 * @returns
 */
function Loading({ type, ...rest }: LoadingProperties) {
  switch (type) {
    case LoadingTypeEnum.circular:
      return <MuiCircularProgress {...rest} disableShrink />;
    default:
      return (
        <MuiContainer maxWidth='lg' sx={{ py: 10 }}>
          <MuiLinearProgress {...rest} />
        </MuiContainer>
      );
  }
}

/**
 * Default values for loading component props
 *
 * @type {{ type: LoadingTypeEnum; }}
 */
const defaultProps = { type: LoadingTypeEnum.linear };

Loading.defaultProps = defaultProps;

export { LoadingTypeEnum };
export default Loading;
