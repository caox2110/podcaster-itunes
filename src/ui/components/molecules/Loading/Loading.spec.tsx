import { render } from '@/core/test';
import type { RenderResult } from '@/core/test';

import Loading, { LoadingTypeEnum } from './Loading';

const properties = {
  type: LoadingTypeEnum.linear,
};

function renderApp({ ...rest }: { type?: LoadingTypeEnum } = properties): RenderResult {
  const utils = render(<Loading {...rest} />, { withRouter: true });

  return {
    ...utils,
  };
}

describe('Loading', () => {
  describe('Rendering', () => {
    it('Should render linear loading without throwing any errors', () => {
      const { baseElement, container } = renderApp();

      expect(baseElement).toBeTruthy();
      expect(container).toMatchSnapshot();
    });
    it('Should render circular loading without throwing any errors', () => {
      const { baseElement, container } = renderApp({ type: LoadingTypeEnum.circular });

      expect(baseElement).toBeTruthy();
      expect(container).toMatchSnapshot();
    });
  });
});
