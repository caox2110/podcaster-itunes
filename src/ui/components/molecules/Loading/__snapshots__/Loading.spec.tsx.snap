// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Rendering Should render circular loading without throwing any errors 1`] = `
<div>
  <span
    class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
    role="progressbar"
    style="width: 40px; height: 40px;"
  >
    <svg
      class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate MuiCircularProgress-circleDisableShrink css-79nvmn-MuiCircularProgress-circle"
        cx="44"
        cy="44"
        fill="none"
        r="20.2"
        stroke-width="3.6"
      />
    </svg>
  </span>
</div>
`;

exports[`Loading Rendering Should render linear loading without throwing any errors 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-lcgnqy-MuiContainer-root"
  >
    <span
      class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-indeterminate css-eglki6-MuiLinearProgress-root"
      role="progressbar"
    >
      <span
        class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate css-5ir5xx-MuiLinearProgress-bar1"
      />
      <span
        class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar2Indeterminate css-1r8wrcl-MuiLinearProgress-bar2"
      />
    </span>
  </div>
</div>
`;
